velox/dwio/parquet/common/BitStreamUtilsInternal.h

```diff
@@ -206,281 +208,276 @@ class BitReader {
 
  private:
   const uint8_t* buffer_;
-  int max_bytes_;
+  int maxBytes_;
 
   /// Bytes are memcpy'd from buffer_ and values are read from this variable.
   /// This is faster than reading values byte by byte directly from buffer_.
-  uint64_t buffered_values_;
+  uint64_t bufferedValues_;
 
-  int byte_offset_; // Offset in buffer_
-  int bit_offset_; // Offset in buffered_values_
+  int byteOffset_; // Offset in buffer_
+  int bitOffset_; // Offset in bufferedValues_
 };
 
-inline bool BitWriter::PutValue(uint64_t v, int num_bits) {
-  DCHECK_LE(num_bits, 64);
-  if (num_bits < 64) {
-    DCHECK_EQ(v >> num_bits, 0) << "v = " << v << ", num_bits = " << num_bits;
+inline bool BitWriter::PutValue(uint64_t v, int numBits) {
+  VELOX_DCHECK_LE(numBits, 64);
+  if (numBits < 64) {
+    VELOX_DCHECK_EQ(v >> numBits, 0, "v = {}, numBits = {}", v, numBits);
   }
 
-  if (ARROW_PREDICT_FALSE(
-          byte_offset_ * 8 + bit_offset_ + num_bits > max_bytes_ * 8))
+  if (FOLLY_UNLIKELY(byteOffset_ * 8 + bitOffset_ + numBits > maxBytes_ * 8))
     return false;
 
-  buffered_values_ |= v << bit_offset_;
-  bit_offset_ += num_bits;
-
-  if (ARROW_PREDICT_FALSE(bit_offset_ >= 64)) {
-    // Flush buffered_values_ and write out bits of v that did not fit
-    buffered_values_ = ::arrow::bit_util::ToLittleEndian(buffered_values_);
-    memcpy(buffer_ + byte_offset_, &buffered_values_, 8);
-    buffered_values_ = 0;
-    byte_offset_ += 8;
-    bit_offset_ -= 64;
-    buffered_values_ =
-        (num_bits - bit_offset_ == 64) ? 0 : (v >> (num_bits - bit_offset_));
+  bufferedValues_ |= v << bitOffset_;
+  bitOffset_ += numBits;
+
+  if (FOLLY_UNLIKELY(bitOffset_ >= 64)) {
+    // Flush bufferedValues_ and write out bits of v that did not fit
+    bufferedValues_ = ::arrow::bit_util::ToLittleEndian(bufferedValues_);
```

Comment:
We can use `folly::Endian::little(bufferedValues_)`.
The goal is to move away from the arrow dependency.